ecssd_agent.go

```diff
@@ -371,13 +408,17 @@ func main() {
 	}
 	configuration.HostedZoneId = zoneId
 	metadataClient := ec2metadata.New(session.New())
-	hostname, err := metadataClient.GetMetadata("/hostname")
-	configuration.Hostname = hostname
+	localIP, err := metadataClient.GetMetadata("/local-ipv4")
+	configuration.Hostname = "ip-" + strings.Replace(localIP, ".", "-", 3) + "." + DNSName
```

Comment:
Is this needed here? In my case, `/hostname` is already of the form "ip-a-b-d-c.domain". Instead, you may only call `createARecord` if the hostname matches this pattern.